crates/primitives/src/transaction/pooled.rs

```diff
@@ -70,7 +62,8 @@ impl PooledTransactionsElement {
         if tx.is_eip4844() {
             return Err(tx)
         }
-        Ok(tx.into())
+        // todo this is bad additional clone, better method?
+        PooledTransactionsElement::try_from(tx.clone()).map_err(|_| tx)
```

Comment:
we can get rid of this clone by flipping the try_from and this impl, so that try_from uses this function and maps the error

Reply:
why we want to return tx in the err here? any specific use case?

Reply:
from caller pov it's more convenient to turn the failed tx in the error, in case you need to handle that, otherwise you'd need to clone if you want error handling